sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -0,0 +1,149 @@
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class CosmosResponseDiagnosticsTest extends TestSuiteBase {
+    private CosmosClient gatewayClient;
+    private CosmosDatabase createdDatabase;
+    private CosmosContainer cosmosContainer;
+    private CosmosClient directClient;
+
+    private final static String DATABASE_NAME = "TestDiagnosticDB";
+    private final static String CONTAINER_NAME = "TestDiagnosticContainer";
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosResponseDiagnosticsTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() throws Exception {
+        assertThat(this.gatewayClient).isNull();
+        ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+        connectionPolicy.setConnectionMode(ConnectionMode.GATEWAY);
+        gatewayClient = clientBuilder().setConnectionPolicy(connectionPolicy).buildClient();
+        createdDatabase = TestSuiteBase.createSyncDatabase(gatewayClient, DATABASE_NAME);
+        CosmosContainerProperties cosmosContainerProperties = new CosmosContainerProperties(CONTAINER_NAME, "/id");
+        cosmosContainer = createdDatabase.createContainer(cosmosContainerProperties).getContainer();
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() throws CosmosClientException {
+        assertThat(this.gatewayClient).isNotNull();
+        createdDatabase.delete();
+        this.gatewayClient.close();
+        if (this.directClient != null) {
+            this.directClient.close();
+        }
+    }
+
+    @Test(groups = {"simple"})
+    public void testGatewayDiagnostics() {
+        CosmosItemProperties cosmosItemProperties = new CosmosItemProperties();
+        cosmosItemProperties.setId(UUID.randomUUID().toString());
+        try {
+            CosmosItemResponse createResponse = cosmosContainer.createItem(cosmosItemProperties);
+            String diagnostics = createResponse.getCosmosResponseDiagnostics().toString();
+            assertThat(diagnostics).contains("Connection Mode : " + ConnectionMode.GATEWAY);
+            assertThat(diagnostics).contains("Gateway statistics");
+            assertThat(diagnostics).contains("Operation Type : " + OperationType.Create);
+            assertThat(createResponse.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+        } catch (CosmosClientException e) {
+            fail(e.getMessage());
```

Comment:
don't catch and fail, if this the test method throws, testng will automatically fail. drop try/catch

Reply:
Done